mosaic/descriptor.py

```diff
@@ -0,0 +1,277 @@
+import numpy as np 
+
+from functools import cached_property
+from xarray.core.dataset import Dataset  
+
+renaming_dict = {"lonCell": "xCell",
+                 "latCell": "yCell",
+                 "lonEdge": "xEdge",
+                 "latEdge": "yEdge",
+                 "lonVertex": "xVertex",
+                 "latVertex": "yVertex"}
+
+connectivity_arrays = ["cellsOnEdge",
+                       "cellsOnVertex",
+                       "verticesOnEdge",
+                       "verticesOnCell"]
+
+class Descriptor:
+    """
+    Class describing unstructured MPAS meshes in order to support plotting
+    within `matplotlib`. The class constains various methods to create 
+    `matplotlib.collections.PolyCollection` objects for variables defined at
+    cell centers, vertices, and edges.
+    
+
+    Attributes
+    ----------
+    latlon : boolean
+        Whethere to use the lat/lon coordinates in patch construction
+
+        NOTE: I don't think this is needed if the projection arg is
+              properly used at initilaization 
+
+    projection : cartopy.crs.CRS
+
+    transform : cartopy.crs.CRS
+
+    cell_patches : np.ndarray
+
+    edge_patches : np.ndarray
+
+    vertex_patches : np.ndarray
+    """
+    def __init__(self, ds, projection=None, transform=None, use_latlon=False): 
```

Comment:
I really like how you've done explicit typing in in `polycolor()`.  A goal would be to do typing like that here in `Descriptor`, too.

Reply:
Absolutely! This `Descriptor` class is a little bit more a work in progress than `polypcolor`, so I've just held off until I have a more polished version. Maybe I'll add explicit typing as one of the goals for `0.2.0`.